src/couch_replicator_connection.erl

```diff
@@ -171,6 +175,33 @@ delete_worker(Worker) ->
     ok.
 
 
+-spec incr_monitor(pid(), #state{}) -> #state{}.
+incr_monitor(Owner, State) ->
+    NewMonitors = case dict:find(Owner, State#state.monitors) of
+        error ->
+            MRef = monitor(process, Owner),
+            dict:store(Owner, {1, MRef}, State#state.monitors);
```

Comment:
This works. But as a minor nit, was wondering if monitoring can be made simpler by associating a monitor with a connection by Ref. Perhaps, that would avoid, having a to maintain a dict and a ref-count because we refer to owners by pid only.

For example:

 * On a acquire/create, before returning we monitor and save ref in ets. Then return.  If owner exits we get the ref from message,  find the worker which matches that ref, delete him.

 * On relinquish, we make sure to demonitor (with a [flush] options perhaps) the ref in ets, and set it to undefined.

 * If worker exits or crashes, demonitor the ref we find it its ets entry.